CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxNodeCasting.swift

```diff
@@ -0,0 +1,68 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+
+@MemberBlockItemListBuilder
+func choiceNodeCastingMethods(for syntaxNodeKind: SyntaxNodeKind) -> MemberBlockItemListSyntax {
+  if syntaxNodeKind.isBase {
+    DeclSyntax(
+      """
+      public func `is`<S: \(syntaxNodeKind.protocolType)>(_ syntaxType: S.Type) -> Bool {
```

Comment:
How about adding to those methods documentation in the same manner as to others e.g. https://github.com/apple/swift-syntax/blob/00dc4d966c25459a95cc65d4cab11013616a2d63/Sources/SwiftSyntax/SyntaxProtocol.swift#L133-L143 ?

Reply:
Sounds good for me

Reply:
Yes, those doc comments would be useful.